src/components/Card/PCard.vue

```diff
@@ -1,32 +1,36 @@
 <template>
-  <div class="p-card" :class="classes">
-    <slot />
-  </div>
+  <CardBase
+    :class="props.class"
+    :variant="props.variant"
+  >
+    <CardHeader v-if="$slots.header || $slots.title || $slots.description">
+      <CardTitle v-if="$slots.title">
+        <slot name="title" />
+      </CardTitle>
+      <CardDescription v-if="$slots.description">
+        <slot name="description" />
+      </CardDescription>
+    </CardHeader>
+    <CardContent>
+      <slot />
+    </CardContent>
+    <CardFooter v-if="$slots.footer">
+      <slot name="footer" />
+    </CardFooter>
+  </CardBase>
 </template>
 
-<script lang="ts" setup>
-  import { computed } from 'vue'
+<script setup lang="ts">
+  import { type HTMLAttributes } from 'vue'
+  import CardBase, { type CardProps } from '@/components/Card/PCardBase.vue'
+  import CardContent from '@/components/Card/PCardContent.vue'
+  import CardDescription from '@/components/Card/PCardDescription.vue'
+  import CardFooter from '@/components/Card/PCardFooter.vue'
+  import CardHeader from '@/components/Card/PCardHeader.vue'
+  import CardTitle from '@/components/Card/PCardTitle.vue'
 
   const props = defineProps<{
-    flat?: boolean,
+    class?: HTMLAttributes['class'],
+    variant?: CardProps['variant'],
   }>()
-
-  const classes = computed(() => ({
-    'p-card--flat': props.flat,
-    'p-background': !props.flat,
```

Comment:
How are you accounting for nested cards? Currently we have a system with p-background so that if you end up with a card inside a card they are not the same color. Obviously this isn't a good ux pattern and we don't do it intentionally. But user generated schema based forms can have this. 

I see now the card has a border around it. Is that sufficient for this?